x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValueSources.java

```diff
@@ -71,19 +78,53 @@ public static List<ValueSourceInfo> sources(
             try {
                 fieldData = ctx.getForField(fieldType, MappedFieldType.FielddataOperation.SEARCH);
             } catch (IllegalArgumentException e) {
-                if (asUnsupportedSource) {
-                    sources.add(
+                switch (elementType) {
+                    case BYTES_REF -> sources.add(
```

Comment:
nit: you could build the `ValueSourceInfo` in the switch and add it to sources outside the switch.